docs/en/guides/pipeline-integrations/development/sgtk-developer-generating-path-and-publish.md

```diff
@@ -0,0 +1,567 @@
+
+```python
+context = tk.context_from_entity("Task", 13155)
+```
+
+If you print out a representation of the context instance you will get something like this:
+```python
+print(repr(context))
+
+>> <Sgtk Context:   Project: {'type': 'Project', 'name': 'My Project', 'id': 176}
+  Entity: {'type': 'Shot', 'name': 'shot01_running_away', 'id': 1381}
+  Step: {'type': 'Step', 'name': 'Comp', 'id': 8}
+  Task: {'type': 'Task', 'name': 'Comp', 'id': 13155}
+  User: None
+  Shotgun URL: https://mysite.shotgunstudio.com/detail/Task/13155
+  Additional Entities: []
+  Source Entity: {'type': 'Shot', 'name': 'shot01_running_away', 'id': 1381}>
+
+```
+
+Even though you only provided the task, it should have filled in the other related details.  
+
+The publish script should now look like this:
+
+```python 
+import sgtk
+
+# Get the engine instance that is currently running.
+current_engine = sgtk.platform.current_engine()
+
+# Grab the pre-created Sgtk instance from the current engine.
+tk = current_engine.sgtk
+
+# Get a context object from a Task, this Task must belong to a Shot for the future steps to work. 
+context = tk.context_from_entity("Task", 13155)
+```
+
+## Part 4 - Creating Folders
+
+Toolkit can dynamically generate a folder structure on disk based upon your project entities.
+
+This fulfills two purposes.
+
+1. You get an organized structure created on disk where you can place your files.
+2. It allows Toolkit to programmatically understand your structure, derive context from it, and know where to place files.
+
+You need to ensure that the folders exist on disk so that you can resolve the path in a later step.
+You will use the [Sgtk.create_filesystem_structure()](https://developer.shotgunsoftware.com/tk-core/core.html?#sgtk.Sgtk.create_filesystem_structure) method achieve this:
+
+```python
+tk.create_filesystem_structure("Task", context.task["id"])
+```
+You can use the context object to get the task id to generate the folders.
+
+Your code should now look like this:
+
+```python
+import sgtk
+
+# Get the engine instance that is currently running.
+current_engine = sgtk.platform.current_engine()
+
+# Grab the pre-created Sgtk instance from the current engine.
+tk = current_engine.sgtk
+
+# Get a context object from a Task, this Task must belong to a Shot for the future steps to work. 
+context = tk.context_from_entity("Task", 13155)
+
+# Create the required folders based upon the task.
+tk.create_filesystem_structure("Task", context.task["id"])
+```
+
+You've now completed all the preparation steps and are ready to move onto generating a path using a template.
+
+## Part 5 - Using a template to build a path
+
+### Generating the path
+
+Whenever you need to know where a file should be placed or found in Toolkit you can use the templates to resolve an absolute path on disk.
+
+The first thing you need to do is get a template instance for the path you wish to generate.
+
+```python
+template = tk.templates["maya_shot_publish"]
+```
```

Comment:
Yeah, good point. Looks to me like the template dict is not a property or method so it can't be documented at present. I'll bring this up with the rest of the team.
I agree linking to it would be good.
For now, I've updated the paragraph to include more description:

>The first thing you need to do is get a template instance for the path you wish to generate. Using the `Sgtk` instance you created, you can access the desired `Template` instance via the `templates` attribute, which is a dictionary where the keys are the template names, and the values are [`Template`](https://developer.shotgunsoftware.com/tk-core/core.html#template) instances.

Reply:
This might need to be the solution to documenting the templates property in core docs. I'll take a look:

https://stackoverflow.com/questions/8649105/how-to-show-instance-attributes-in-sphinx-doc

Reply:
This change to the core API is now in QA